### PR TITLE
update unified pipeline script to support E2E flow

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK

- Update Unified pipeline script to use .Net E2E flow

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/scripts/Cadl-Project-Generate.ps1 [hunk 1]
```diff
@@ -42,6 +42,17 @@ $npmWorkingDir = Resolve-Path $tempFolder/$innerFolder
 $mainCadlFile = If (Test-Path "$npmWorkingDir/client.cadl") { Resolve-Path "$npmWorkingDir/client.cadl" } Else { Resolve-Path "$npmWorkingDir/main.cadl"}
 $resolvedProjectDirectory = Resolve-Path "$ProjectDirectory/src"
 
+if (Test-Path "$npmWorkingDir/cadl-project.yaml") {
+    $cadlProjectYaml = Get-Content -Path "$npmWorkingDir/cadl-project.yaml" -Raw
+
+    Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
```
Comment (benbp) [inline eng/scripts/Cadl-Project-Generate.ps1:48]: This should be removed, we don't want/need to check for the yaml module multiple times in the script.

File: eng/scripts/Cadl-Project-Generate.ps1 [hunk 2]
```diff
@@ -42,6 +42,17 @@ $npmWorkingDir = Resolve-Path $tempFolder/$innerFolder
...
+    $yml = ConvertFrom-YAML $cadlProjectYaml
+    if ( $yml["emitters"]["@azure-tools/cadl-csharp"]["sdk-folder"]) {
+        $yml["emitters"]["@azure-tools/cadl-csharp"]["sdk-folder"] = ""
+    }
+    (ConvertTo-Yaml $yml) | Out-File "$npmWorkingDir/cadl-project.yaml"
```
Comment (weshaggard) [inline eng/scripts/Cadl-Project-Generate.ps1:53]: Do we need this work at all? Or is this just temporary until the cadl team fixes their output parameter support?

Comment (chunyu3) [reply eng/scripts/Cadl-Project-Generate.ps1:53]: @weshaggard  Yes. it is temporary. And will be removed when cadl support output parameter.

File: eng/scripts/Cadl-Project-Generate.ps1 (hunk 1 above)
Comment (chunyu3) [reply eng/scripts/Cadl-Project-Generate.ps1]: Thanks @benbp I removed it.

**Review by dw511214992 — APPROVED** [review]
Looks good to me, and the test runs succesfully in test env.

**Pull request merged**